### ISSUE TITLE
drawio macro cannot be added to a draft Confluence Cloud page

### ISSUE THREAD

**Issue by mararad** [jgraph/drawio#4095]
Reported in https://community.atlassian.com/t5/Confluence-questions/Can-t-add-draw-io-widgets-to-saved-draft/qaq-p/2572141 
Reproduced in the latest Chrome. Gliffy has the same issue (with one difference).

When I try to add drawio macro to Confluence draft page, it immediately shows the error "Draft Write Error: Draft could not be created" (drawio macro could not be added) and HAR shows "com.atlassian.confluence.api.service.exceptions.NotFoundException: No content found with id". PUT error 404
When I repeated the steps with Gliffy, Gliffy macro could be added but the error appeared on Save

Glifffy tested on this page:
https://draw-test2.atlassian.net/wiki/spaces/DT2/pages/edit-v2/2219376641?draftShareId=f92fec00-ddc9-4512-9cf3-affc5f3a560a
![404](https://github.com/jgraph/drawio/assets/44292306/412e7909-488d-40fc-806f-cd8443d5b211)

![gliffy draft error](https://github.com/jgraph/drawio/assets/44292306/6e049340-051b-49b0-b54d-8a9fb98e7ff6)
![headers](https://github.com/jgraph/drawio/assets/44292306/cda9400d-1bbd-47dc-a56a-9d84af47962f)
[drawio.atlassian.net-1704880932448.log](https://github.com/jgraph/drawio/files/13886767/drawio.atlassian.net-1704880932448.log)

https://github.com/jgraph/drawio/assets/44292306/dfd1527d-b49b-4745-b134-ddd601a7dd78

**Comment by m-mohamedin** [comment]
I cannot reproduce currently. Maybe a temporary error with Confluence
@mararad can you please re-test

**Comment by mararad** [comment]
@m-mohamedin On the first re-test I couldn't recreate the issue, all was fine. 
On the 2nd retest, I recreated the issue and also saw board macro has issue on save. 
![test board on draft](https://github.com/jgraph/drawio/assets/44292306/eecf32ea-c327-40f7-b3d5-7ccd77fb919e)
Then I cleared browser cache and tested for the 3rd time and the issue was gone, all was good again.

My conclusion at this moment would be: the issue reappears on every new draft open and can be solved by cache clear but this is only one time solution because the issue appears again. Here is my test draft page: https://drawio.atlassian.net/wiki/spaces/DT2/pages/edit-v2/4257480710?draftShareId=19b93fc0-6cfd-4339-a282-c07df7c2b100 where I managed to add new diagrams and templates only after I cleared cache.

**Comment by thesuavehog** [comment]
I just added draw.io to my Confluence (cloud) and I am experiencing this issue as well.
When I try to add a `/draw` board to my Confluence page, the draw.io UX opens but when I click 'Publish' I get...

1. First click on Publish it asks for a name for the file and then shows an error and a constant spinner
2. Second click on Publish it shows a dialog now with not just filename but also `type` and `where` input fields
   * when I click Save then I get the same error as the previous commenter shows about the `showSaveFilePicker`
 
I have tried in Edge and Chrome on Windows (latest version of each).
* Edge: Version 123.0.2420.81 (Official build) (64-bit)
* Chrome: Version 123.0.6312.106 (Official Build) (64-bit)
